### PR TITLE
Added support for `-derivedDataPath` and `-resultBundlePath` options.

### PR DESCRIPTION
**Pull request by bartoszj**
Hi,

Sometimes for Jenkins jobs I need to set derived data folder and/or set result bundle path. I've added additional options to handle `-derivedDataPath` and `-resultBundlePath`.

Also I've added unit tests for `Scan::TestCommandGenerator` which are based on `Gym:BuildCommandGenerator`.

### CODE REVIEW

File: lib/scan/options.rb [hunk 1]
```diff
@@ -73,6 +73,16 @@ def self.available_options
                                      env_name: "SCAN_BUILDLOG_PATH",
                                      description: "The directory were to store the raw log",
                                      default_value: "~/Library/Logs/scan"),
+        FastlaneCore::ConfigItem.new(key: :derived_data_path,
+                                     short_option: "-e",
+                                     env_name: "SCAN_DERIVED_DATA_PATH",
+                                     description: "The directory where build products and other derived data will go",
+                                     optional: true),
+        FastlaneCore::ConfigItem.new(key: :result_bundle,
```
Comment (KrauseFx) [inline lib/scan/options.rb:81]: Could you also add a `short_option` to allow passing it via the CLI :+1:

File: lib/scan/options.rb [hunk 2]
```diff
@@ -73,6 +73,16 @@ def self.available_options
                                      env_name: "SCAN_BUILDLOG_PATH",
                                      description: "The directory were to store the raw log",
                                      default_value: "~/Library/Logs/scan"),
+        FastlaneCore::ConfigItem.new(key: :derived_data_path,
+                                     short_option: "-e",
```
Comment (KrauseFx) [inline lib/scan/options.rb]: Could you change away from `-e`, I just recently added this short option to another option

**Comment by KrauseFx** [comment]
Looks good to me, I added some comments before it's ready. cc @mpirri

**Comment by bartoszj** [comment]
@KrauseFx. It should be OK now.

**Comment by KrauseFx** [comment]
Thanks, that's awesome @bartoszj :+1:

**Pull request merged**